cpp/src/strings/replace/multi_re.cu

```diff
@@ -149,22 +148,26 @@ std::unique_ptr<column> replace_re(
   auto repls_column   = column_device_view::create(repls.parent(), stream);
   auto d_repls        = *repls_column;
   auto d_flags        = get_character_flags_table();
+
   // compile regexes into device objects
   size_type regex_insts = 0;
   std::vector<std::unique_ptr<reprog_device, std::function<void(reprog_device*)>>> h_progs;
-  rmm::device_vector<reprog_device> progs;
+  thrust::host_vector<reprog_device> progs;
   for (auto itr = patterns.begin(); itr != patterns.end(); ++itr) {
-    auto prog  = reprog_device::create(*itr, d_flags, strings_count, stream);
-    auto insts = prog->insts_counts();
-    if (insts > regex_insts) regex_insts = insts;
+    auto prog   = reprog_device::create(*itr, d_flags, strings_count, stream);
+    regex_insts = std::max(regex_insts, prog->insts_counts());
     progs.push_back(*prog);
     h_progs.emplace_back(std::move(prog));
   }
-  auto d_progs = progs.data().get();
 
-  // copy null mask
-  auto null_mask  = copy_bitmask(strings.parent());
-  auto null_count = strings.null_count();
+  // copy all the reprog_device instances to a device memory array
+  rmm::device_buffer progs_buffer{sizeof(reprog_device) * progs.size()};
+  cudaMemcpy(progs_buffer.data(),
+             progs.data(),
+             progs.size() * sizeof(reprog_device),
+             cudaMemcpyHostToDevice);
```

Comment:
```suggestion
  cudaMemcpyAsync(progs_buffer.data(),
             progs.data(),
             progs.size() * sizeof(reprog_device),
             cudaMemcpyHostToDevice,
             stream.value());
```

Reply:
Ah good catch. I need to wrap that in a `CUDA_TRY` as well.